### PR TITLE
CVE-2009-2659

### PR DESCRIPTION
**Pull request by spc7641**
Updated all fields for CVE-2009-2659, a vulnerability about improper validation of input urls.

### CODE REVIEW

**Comment by Scar56** [comment]
the date discovered is after the date reported/annonuced/published, take another look.
good job otherwise
upvotes 3

File: cves/CVE-2009-2659.yml [hunk 1]
```diff
@@ -15,24 +15,24 @@ curated_instructions: |
...
   source for this is Chrome's Stable Release Channel
   (https://chromereleases.googleblog.com/).
   Please enter your date in YYYY-MM-DD format.
-announced_date:
+announced_date: 2009-07-28
 published_instructions: |
   Is there a published fix or patch date for this vulnerability?
   Please enter your date in YYYY-MM-DD format.
-published_date:
+published_date: 2009-07-28
```
Comment (ADAdebomi) [inline cves/CVE-2009-2659.yml]: Same dates for published, announced and reported. Possible mistake?

File: cves/CVE-2009-2659.yml [hunk 2]
```diff
@@ -270,7 +279,11 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    The mistakes here were coding mistakes that included not adding proper checks for the url and found files
+    as well as not including proper error handling. Instead of using methods such as safe_join to create the url that would be tested
+    the original programmers simply used a join on any inputs, and no error handling was put in place for if the file path was invalid. 
+    This could have been fixed by having a better code reviews that would have identified that the lack of error handling and sanitization of urls.
```
Comment (ADAdebomi) [inline cves/CVE-2009-2659.yml:288]: Great job on the mistakes and explaining it, also providing a mitigation

File: cves/CVE-2009-2659.yml [hunk 3]
```diff
@@ -49,7 +49,10 @@ description_instructions: |
 
   Your target audience is people just like you before you took any course in
   security
-description:
+description: |
```
Comment (ADAdebomi) [inline cves/CVE-2009-2659.yml:54]: Upvote:4

**Pull request merged**